### PR TITLE
Remove integration build tag

### PR DESCRIPTION
**Pull request by guillaumerose**
Simplify the build code and fix lint errors.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -121,6 +120,9 @@ clean: clean_docs
 	rm -rf $(RELEASE_DIR)
        
 
+$(HOST_BUILD_DIR)/integration.test: $(SOURCES)
```
Comment (praveenkumar) [inline Makefile:123]: What is the use of this target, may be put that details to commit log also?

File: Makefile [hunk 2]
```diff
@@ -96,7 +95,7 @@ cross: $(BUILD_DIR)/macos-amd64/crc $(BUILD_DIR)/linux-amd64/crc $(BUILD_DIR)/wi
 
 .PHONY: test
 test:
-	go test --tags build -v -ldflags="$(VERSION_VARIABLES)" $(shell $(PACKAGES))
+	go test --tags build -v -ldflags="$(VERSION_VARIABLES)" ./pkg/... ./cmd/...
```
Comment (praveenkumar) [inline Makefile:98]: :+1:

File: Makefile (hunk 1 above)
Comment (guillaumerose) [reply Makefile:123]: I forgot to add it but we should check we can build integration test on all platforms. Perhaps in a new target like `make cross-integration`.

Comment (praveenkumar) [reply Makefile:123]: @guillaumerose but then crc binary for mac/windows not going to run on linux so `make cross-integration` will not work on a single platform.

Comment (guillaumerose) [reply Makefile]: I added the CI target, let me know what you think of https://github.com/code-ready/crc/pull/1437/commits/fa753f28b43688cf69ac09b78a107726bddff89e !

File: Makefile [hunk 3]
```diff
@@ -119,7 +118,12 @@ clean: clean_docs
 	rm -rf $(BUILD_DIR)
 	rm -f $(GOPATH)/bin/crc
 	rm -rf $(RELEASE_DIR)
-       
+
+.PHONY: build_integration
+build_integration: $(SOURCES)
+	GOOS=linux   go test ./test/integration/ -c -o $(BUILD_DIR)/linux-amd64/integration.test
+	GOOS=windows go test ./test/integration/ -c -o $(BUILD_DIR)/windows-amd64/integration.test.exe
+	GOOS=darwin  go test ./test/integration/ -c -o $(BUILD_DIR)/darwin-amd64/integration.test
```
Comment (praveenkumar) [inline Makefile:126]: @guillaumerose This is really new for me, I didn't know we can even do that :(. can this binary used directly starting the integration tests with the arguments?

Comment (praveenkumar) [reply Makefile:126]: @guillaumerose Also for mac target we use `macos-amd64` instead `darwin-amd64`

Comment (guillaumerose) [reply Makefile]: Fixed macos/darwin mistake.

Yes we can use the binary! The only issue I have is that it uses `os.Getwd` and it really wants to be launched in `test/integration`.

The Makefile should look like

``` 
integration: 
	(cd ./test/integration && $(BUILD_DIR)/linux-amd64/integration.test --test.timeout=120m -test.v $(PULL_SECRET_FILE) $(BUNDLE_LOCATION) $(CRC_BINARY) --bundle-version=$(BUNDLE_VERSION) $(GODOG_OPTS))
```

I will open a new PR with this change, if it's ok for you :)

Comment (praveenkumar) [reply Makefile]: Ah so it is only create the binary which doesn't include feature files and that's the reason needed to launched from `test/integration` ?

Comment (guillaumerose) [reply Makefile]: yes!

**Pull request merged**